addons/website_payment/static/src/js/payment_transaction.js

```diff
@@ -3,11 +3,26 @@ odoo.define('website_payment.website_payment', function (require) {
 
 var website = require('website.website');
 var ajax = require('web.ajax');
+var Dialog = require('web.Dialog');
 
 if(!$('.o_website_payment').length) {
     return $.Deferred().reject("DOM doesn't contain '.o_website_payment'");
 }
 
+$('.delete_payment').on('click', function(e){
+      var $this = $(this);
+      e.preventDefault();
+      console.log("ok");
+      Dialog.confirm(null,
+        "Be aware that this credit card is linked to an ongoing contract, it will prevent the payment of it. Are you sure you want to proceed?",
+        {'confirm_callback': function(result){
+          ajax.post($this.attr('href'),{}).done(function(){
+            location.reload();
```

Comment:
do you need location.reload, while delete controller already redirecting to /my/payment_method